datasafe-rest/src/main/java/de/adorsys/datasafe/rest/controller/UserController.java

```diff
@@ -0,0 +1,82 @@
+package de.adorsys.datasafe.rest.controller;
+
+import de.adorsys.datasafe.business.api.types.CreateUserPrivateProfile;
+import de.adorsys.datasafe.business.api.types.CreateUserPublicProfile;
+import de.adorsys.datasafe.business.api.types.UserID;
+import de.adorsys.datasafe.business.api.types.UserIDAuth;
+import de.adorsys.datasafe.business.api.types.keystore.ReadKeyPassword;
+import de.adorsys.datasafe.business.api.types.resource.*;
+import de.adorsys.datasafe.business.impl.service.DefaultDatasafeServices;
+import de.adorsys.datasafe.rest.dto.UserCreateRequestDTO;
+import io.swagger.annotations.ApiOperation;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@RestController()
+@RequestMapping(value = "/user", produces = APPLICATION_JSON_VALUE, consumes = APPLICATION_JSON_VALUE)
+public class UserController {
+
+    @Value("${datasafe.system.root}")
+    private String systemRoot;
+
+    private static final String PRIVATE_COMPONENT = "private";
+    private static final String PRIVATE_FILES_COMPONENT = PRIVATE_COMPONENT + "/files";
+    private static final String INBOX_COMPONENT = "inbox";
+
+    private final DefaultDatasafeServices dataSafeService;
+
+    public UserController(DefaultDatasafeServices dataSafeService) {
+        this.dataSafeService = dataSafeService;
+    }
+
+    @PostMapping("/")
+    @ApiOperation("Create user")
+    public void createUser(@RequestBody UserCreateRequestDTO requestDTO) {
+
+        ReadKeyPassword readKeyPassword = new ReadKeyPassword(requestDTO.getPassword());
+        UserIDAuth auth = new UserIDAuth(new UserID(requestDTO.getUserName()), readKeyPassword);
+
+        URI rootLocation = URI.create(systemRoot);
+        rootLocation = rootLocation.resolve(requestDTO.getUserName() + "/");
+        URI inboxUri = rootLocation.resolve("./" + INBOX_COMPONENT + "/");
+
+        URI keyStoreUri = rootLocation.resolve("./" + PRIVATE_COMPONENT + "/keystore");
+
+        dataSafeService.userProfile().registerPublic(CreateUserPublicProfile.builder()
+                .id(auth.getUserID())
+                .inbox(access(inboxUri))
+                .publicKeys(access(keyStoreUri))
+                .build()
+        );
+
+        URI filesUri = rootLocation.resolve("./" + PRIVATE_FILES_COMPONENT + "/");
+
+        dataSafeService.userProfile().registerPrivate(CreateUserPrivateProfile.builder()
```

Comment:
Please check develop, this code is outdated

Reply:
ok